berkdb/db/db_am.c

```diff
@@ -109,10 +122,17 @@ __db_cursor_int(dbp, txn, dbtype, root, is_opd, lockerid, dbcp, flags)
 			 * is running concurrently using this DB, so we can
 			 * take a peek at any cursors on the active queue.
 			 */
-			if (!DB_IS_THREADED(dbp) &&
-			    (adbc = TAILQ_FIRST(&dbp->active_queue)) != NULL)
-				dbc->lid = adbc->lid;
-			else {
+			if (!DB_IS_THREADED(dbp)) {
+				if (!dbp->use_tlcq)
+					adbc = TAILQ_FIRST(&dbp->active_queue);
+				else {
+					cq = __db_acquire_cq(dbp);
```

Comment:
Can __db_acquire_cq return NULL here?

Reply:
Yes. It can return NULL. I've added a check for that.